### PR TITLE
Clean strings usages in CMake.

### PR DESCRIPTION
**Pull request by bosilca**
When we compare a CMake variable to a string, there is no need to pass
it as a string (aka "${}"), we can just use it directly as a named
variable.

Signed-off-by: George Bosilca <bosilca@icl.utk.edu>

### CODE REVIEW

File: cmake_modules/ParsecCompilerFlags.cmake [hunk 1]
```diff
@@ -187,9 +186,9 @@ endif()
 if(CMAKE_Fortran_COMPILER_WORKS)
   get_filename_component (Fortran_COMPILER_NAME ${CMAKE_Fortran_COMPILER} NAME)
   #  message(STATUS "Fortran Compiler ${Fortran_COMPILER_NAME} id is ${CMAKE_Fortran_COMPILER_ID}")
-  if(${CMAKE_Fortran_COMPILER_ID} STREQUAL "GNU")
+  if(CMAKE_Fortran_COMPILER_ID STREQUAL "GNU")
     # gfortran or g77
-    if(${Fortran_COMPILER_NAME} MATCHES g77)
+    if("${Fortran_COMPILER_NAME}" MATCHES g77)
```
Comment (devreal) [inline cmake_modules/ParsecCompilerFlags.cmake:191]: Looks to me as if `MATCHES` works with variable names too: https://cmake.org/cmake/help/latest/command/if.html#variable-expansion

File: parsec/parsec.c [hunk 2]
```diff
@@ -701,11 +698,17 @@ parsec_context_t* parsec_init( int nb_cores, int* pargc, char** pargv[] )
 #if defined(PARSEC_PROF_TRACE)
     if( (0 != strncasecmp(parsec_enable_profiling, "<none>", 6)) && (0 == parsec_profiling_init( profiling_id )) ) {
         int i, l;
-        char *cmdline_info = basename(parsec_app_name);
+        char *cmdline_info = NULL;
 
         /* Use either the app name (argv[0]) or the user provided filename */
         if( 0 == strncmp(parsec_enable_profiling, "<app>", 5) ) {
+            /* Specialize the profiling filename to avoid collision with other instances */
```
Comment (devreal) [inline parsec/parsec.c:705]: This seems unrelated and seems to supersede https://github.com/ICLDisco/parsec/pull/339/files. On purpose?

**Review by abouteiller — COMMENTED** [review]
There is also an instance in FindCython that got missed

File: parsec/parsec.c (hunk 2 above)
Comment (bosilca) [reply parsec/parsec.c:705]: gosh, I had a dirty tree so I mixes this PR with the patch I made for profiling issue.

**Review by devreal — APPROVED** [review]

File: parsec/fortran/CMakeLists.txt [hunk 3]
```diff
@@ -1,8 +1,7 @@
 include(FortranCInterface)
 FortranCInterface_HEADER(f2c_mangle.h
                              MACRO_NAMESPACE "PARSEC_F2C_"
-                             SYMBOL_NAMESPACE "parsec_f2c_"
-                             SYMBOLS my_sub parsecf:my_sub)
```
Comment (devreal) [inline parsec/fortran/CMakeLists.txt:5]: Why was this line removed?

Comment (bosilca) [reply parsec/fortran/CMakeLists.txt:5]: Because it was incorrect, we were not providing the my_sub symbol.

**Review by devreal — APPROVED** [review]

**Review by abouteiller — APPROVED** [review]
